stackstac/prepare.py

```diff
@@ -143,6 +143,15 @@ def prepare_items(
             asset_bbox = asset.get("proj:bbox", item_bbox)
             asset_shape = asset.get("proj:shape", item_shape)
             asset_transform = asset.get("proj:transform", item_transform)
+            raster_bands = asset.get('raster:bands')
```

Comment:
If we're going to look at `raster:bands`, it would be good to take advantage of the other metadata too: https://github.com/gjoseph92/stackstac/issues/63#issuecomment-868726296. I think this PR is still valuable without also considering `data_type` and `nodata`, but that would be a nice follow-up.

Reply:
Yeah definitely something to consider